services/secondary-storage/server/src/main/java/org/apache/cloudstack/storage/resource/NfsSecondaryStorageResource.java

```diff
@@ -2050,6 +2050,14 @@ protected Answer deleteSnapshot(final DeleteCommand cmd) {
                 s_logger.warn(details);
                 return new Answer(cmd, false, details);
             }
+
+            // delete the directory if it is empty
+            if (snapshotDir.isDirectory() && snapshotDir.list().length == 0 && !snapshotDir.delete()) {
+                    details = String.format("Unable to delete directory [%s] at path [%s].", snapshotDir.getName(), snapshotPath);
+                    s_logger.debug(details);
+                    return new Answer(cmd, false, details);
+                }
+            }
```

Comment:
```suggestion
                details = String.format("Unable to delete directory [%s] at path [%s].", snapshotDir.getName(), snapshotPath);
                s_logger.debug(details);
                return new Answer(cmd, false, details);
            }
```

@vishesh92, some adjusts are required because the previous suggestions were directly applied.